.github/workflows/integration-test.yml

```diff
@@ -43,13 +45,51 @@ jobs:
   BuildAndUpload:
     uses: ./.github/workflows/test-build.yml
     secrets: inherit
+    permissions:
+      id-token: write
+      contents: read
+    with:
+      BucketKey: "integration-test/binary/${{ github.sha }}"
+      PackageBucketKey: "integration-test/packaging/${{ github.sha }}"
+      TerraformAWSAssumeRole: ${{ vars.TERRAFORM_AWS_ASSUME_ROLE }}
+      Bucket: ${{ vars.S3_INTEGRATION_BUCKET }}
+
+  BuildAndUploadPackages:
```

Comment:
Itar only requires linux packages which comes from the build and upload step so split this out